src/pages/api/submitPaymentTx/index.ts

```diff
@@ -41,9 +43,29 @@ export default async function handler(
     return res.status(500).json({ error: "Unsupported chain" });
   }
 
-  const provider = new ethers.providers.JsonRpcProvider(sponsoredInfo.rpc);
+  if (additionalPayload) {
+    const { txHash } = await relayPayWithAuthorization({ 
+      cartParams: additionalPayload.cartParams,
+      slicerId: additionalPayload.slicerId,
+      totalUsdcPrice: value,
+      buyer: from,
+      chainId,
+      signature,
+      authorizationParams: {
+        from,
+        nonce,
+        to,
+        validAfter,
+        validBefore,
+        value,
+      },
+    });
```

Comment:
As discussed during Sync: Let's move this to a new endpoint called `submitSlicePaymentTx`